repos/fdbt-site/src/pages/additionalPricingStructures.tsx

```diff
@@ -0,0 +1,186 @@
+                                                    className="govuk-input govuk-input--width-3"
+                                                    id="structure-discount"
+                                                    name="structureDiscount"
+                                                    type="text"
+                                                    defaultValue={additionalPricingStructures.structureDiscount}
+                                                />
+                                            </FormElementWrapper>
+                                            <div className="govuk-input__suffix" aria-hidden="true">
+                                                %
+                                            </div>
+                                        </div>
+                                    </div>
+                                </>
+                            </div>
+
+                            <div className="govuk-radios__item">
+                                <input
+                                    className="govuk-radios__input"
+                                    id="no-additional-discounts"
+                                    name="additionalDiscounts"
+                                    type="radio"
+                                    value="no"
+                                    data-aria-controls="conditional-no-additional-discounts"
+                                    defaultChecked={errors.length > 0 && !clickedYes}
+                                />
+                                <label className="govuk-label govuk-radios__label" htmlFor="no-choice">
+                                    No
+                                </label>
+                            </div>
+                        </div>
+                        <br />
+                        <input
+                            type="submit"
+                            value="Continue"
+                            id="continue-button"
+                            className="govuk-button govuk-!-margin-right-2"
+                        />
+                    </fieldset>
+                </>
+            </CsrfForm>
+        </FullColumnLayout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: SelectTimeRestrictionsProps } => {
+    const csrfToken = getCsrfToken(ctx);
+    const additionalPricingStructures:
+        | AdditionalPricing
+        | { clickedYes: boolean; additionalPricingStructures: WithErrors<AdditionalPricing> }
+        | undefined = getSessionAttribute(ctx.req, ADDITIONAL_PRICING_ATTRIBUTE);
+
+    let clickedYes = false;
+    let errors: ErrorInfo[] = [];
+    let additionalPricing: AdditionalPricing = {
+        pricingStructureStart: '',
+        structureDiscount: '',
+    };
+    if (
+        additionalPricingStructures &&
+        'additionalPricingStructures' in additionalPricingStructures &&
```

Comment:
if it is with errors, doesnt that mean 'additionalPricingStructures' in additionalPricingStructures isnt needed? Because we know it will be?